lib/cylc/conditional_simplifier.py

```diff
@@ -100,31 +101,40 @@ def nest_by_oper( self, nest_me, oper ):
         else:
             return nest_me
 
-    def clean_expr( self, nested_list, criteria ):
-        """Return a list with entries specified by 'critria' removed"""
+    def clean_expr( self, nested_list, criterion ):
+        """Return a list with entries specified by 'criterion' removed"""
         cleaned = copy.deepcopy( nested_list )
+
         # Make sure that we don't have extraneous nesting.
         while (isinstance(cleaned, list) and len(cleaned) == 1 and
                isinstance(cleaned[0], list)):
             cleaned = cleaned[0]
 
-        # Recurse through the nested list and remove criteria.
-        found = None
-        if isinstance(cleaned, str) or len(cleaned)==1:
-            if cleaned == criteria:
+        if len(cleaned) == 1:
+            cleaned = cleaned[0]
+
+        if isinstance(cleaned, str):
+            if cleaned == criterion:
```

Comment:
This seemed to be a bug - if `cleaned` is a list with one element, can't go `"abc" == ["abc"]`.